apps/src/redux/urlHelpers.js

```diff
@@ -0,0 +1,33 @@
+/** @file Redux module of helpers for storing and constructing pegasus/dashboard urls */
+
+//Get/Set Pegasus host (proto, domain, port)
+
+// Actions
+const SET_PEGASUS_HOST = 'urlHelpers/SET_PEGASUS_HOST';
+
+const INITIAL_STATE = {
+  pegasusHost: 'https://code.org',
+};
+
+// Reducer
+export default function reducer(state = INITIAL_STATE, action = {}) {
+  switch (action.type){
+    case SET_PEGASUS_HOST:
+      return {pegasusHost: action.hostname};
+    default:
+      return state;
+  }
+}
+
+// Action Creators
+export function setPegasusHost(hostname) {
+  return { type: SET_PEGASUS_HOST, hostname };
+}
+
+// Selectors
+export function pegasusUrl(state, relativePath) {
+  if (state && state.urlHelpers) {
+    return `${state.urlHelpers.pegasusHost}${relativePath}`;
+  }
+  return relativePath;
+}
```

Comment:
I've tried to have my selectors receive the state of the subtree as input rather than the state of the global tree. This way my "ducks" feel more modular, as they don't need to know anything about the shape of the global state tree. The caller would then be responsible for getting to the subtree, i.e.
```
// instead of
pegasusUrl(state, '/rel/path');
// we have
pegasusUrl(state.urlHelpers, '/rel/path');